### PR TITLE
[NFC][AMDGPU][GlobalISel] Test bufer_load/store of pointers (crashes)

### PR DESCRIPTION
**Pull request by krzysz00**

### CODE REVIEW

File: llvm/test/CodeGen/AMDGPU/GlobalISel/buffer-load-store-pointers.ll [hunk 1]
```diff
@@ -0,0 +1,71 @@
+; RUN: not --crash llc -global-isel -mtriple=amdgcn -mcpu=gfx90a -verify-machineinstrs -stop-after=instruction-select < %s
+
+define ptr addrspace(0) @buffer_load_p0(ptr addrspace(8) inreg %buf) {
+  %ret = call ptr addrspace(0) @llvm.amdgcn.raw.ptr.buffer.load.p0(ptr addrspace(8) inreg %buf, i32 0, i32 0, i32 0)
+  ret ptr addrspace(0) %ret
+}
+
+define void @buffer_store_p0(ptr addrspace(0) %data, ptr addrspace(8) inreg %buf) {
+  call void @llvm.amdgcn.raw.ptr.buffer.store.p0(ptr addrspace(0) %data, ptr addrspace(8) inreg %buf, i32 0, i32 0, i32 0)
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/GlobalISel/buffer-load-store-pointers.ll:9]: Remove the addrspace(0)s everywhere

File: llvm/test/CodeGen/AMDGPU/GlobalISel/buffer-load-store-pointers.ll [hunk 2]
```diff
@@ -0,0 +1,71 @@
+; RUN: not --crash llc -global-isel -mtriple=amdgcn -mcpu=gfx90a -verify-machineinstrs -stop-after=instruction-select < %s
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/GlobalISel/buffer-load-store-pointers.ll:1]: Generally don't precommit tests that fail. Also don't need -verify-machineinstrs

Comment (krzysz00) [reply llvm/test/CodeGen/AMDGPU/GlobalISel/buffer-load-store-pointers.ll:1]: Good to know about not precommiting tests that crash - I remember way back in the day someone wanting a `not --crash` test for a bug I was fixing, hence the precommit

Comment (arsenm) [reply llvm/test/CodeGen/AMDGPU/GlobalISel/buffer-load-store-pointers.ll:1]: Sometimes it makes sense, but I don't think it does here

**Pull request closed without merging**